### PR TITLE
Add support for additional RHEL Edge image types

### PR DESCRIPTION
**Pull request by jkozol**
A user can now select the `RHEL for Edge Container (.tar)` image type. This image type supports the optional ostree fields for a parent commit, a ref for the name of the branch, and a url for the parent repository.

The user can also select the `RHEL for Edge Installer (.iso)` image type. This image type supports the required ostree fields of a ref for the name of the branch and a url for the parent repository.

The help text for the new URL field is `Provide the URL of the upstream repository. This is where the OSTree commit will be pulled from.` Does this explain what the URL represents?

There are currently no tests for this PR because these image types are not yet supported by osbuild-composer. They will not display for selection until they are supported.

### CODE REVIEW

**Comment by jkozol** [comment]
Ignore the selected image type. I had to fake it to check the fields.


rheledgeinstaller
rheledgecontainer

**Comment by teg** [comment]
This is great! thanks @jkozol!
I wonder about the second screen, should we make the label `Parent Repository URL`, to make it clear that this is where the parent is taken from, and that we are not actually pushing the container anywhere. Could also put the `Parent commit` box under the `Parent Repository URL` one and in the help text say that it will be inferred from the parent repository if left blank.

**Comment by teg** [comment]
The help text for the `Parent Repository URL` should probably be something like `Provide the URL of the upstream repository. This is where the OSTree commit will be pulled from.` We don't strictly speaking pull in the commit (yet), but this is an implementation detail, and we very well might do in the future.

**Comment by jkozol** [comment]
@teg Would `Parent repo URL` be descriptive enough? There are some limits on how long the labels can be before they wrap and I would prefer to not override patternfly's default styling for this one.

**Comment by teg** [comment]
> @teg Would `Parent repo URL` be descriptive enough? There are some limits on how long the labels can be before they wrap and I would prefer to not override patternfly's default styling for this one.

Maybe just keep it as is and just update the help text?

**Comment by jkozol** [comment]
@teg  `Provide the URL of the upstream repository. This parent repository is where the OSTree commit will be pulled from.`?

**Comment by teg** [comment]
> @teg `Provide the URL of the upstream repository. This parent repository is where the OSTree commit will be pulled from.`?

I think `This repository is where the parent OSTree commit will be pulled from.`

**Review by teg — APPROVED** [review]
Thanks for this!

File: components/Wizard/CreateImageUpload.js [hunk 1]
```diff
@@ -316,7 +338,13 @@ class CreateImageUploadModal extends React.Component {
   }
 
   requiresImageSize(imageType) {
-    if (imageType === "" || imageType === "fedora-iot-commit" || imageType === "rhel-edge-commit") {
+    if (
+      imageType === "" ||
+      imageType === "fedora-iot-commit" ||
+      imageType === "rhel-edge-commit" ||
+      imageType === "rhel-edge-container" ||
+      imageType === "rhel-edge-installer"
```
Comment (gicmo) [inline components/Wizard/CreateImageUpload.js:346]: I wonder if we should not just add `fedora-iot-{container, installer}` as well?

**Comment by jkozol** [comment]
Closing as this has diverged uncomfortably far from master and rebasing would be more painful than starting off of master.

**Pull request closed without merging**